features/step_definitions/logging.rb

```diff
@@ -1056,3 +1056,125 @@
     | pod_label       | #{pod_label}     |
   })
 end
+
+
+
+Given /^I have(?: "(\w+)")? log pod in project #{QUOTED}$/ do | log_type, project_name |
+    log_type ||= "any"
+    file_dir = "#{BushSlicer::HOME}/testdata/logging/loggen"
+
+    unless project(project_name).exists?
+        step %Q/I run the :new_project client command with:/,table(%{
+          | project_name | #{project_name} |
+        })
+        raise "Error creating namespace" unless @result[:success]
+    end
+
+    case log_type
+      when "json"
+        template_file = "#{file_dir}/container_json_log_template.json"
+      when "flat"
+        template_file = "#{file_dir}/container_json_unicode_log_template.json"
+      when "unicode"
+        template_file = "#{file_dir}/container_json_unicode_log_template.json"
+      else
+        template_file = "#{file_dir}/container_json_log_template.json"
+      end
+
+    unless  project.pods(by:user).size > 0 
+        step %Q/I run the :new_app client command with:/,table(%{
+           | file | #{template_file} |
+         })
+    end
+end
+
+Given /^I have index pattern #{QUOTED}$/ do | pattern_name |
+    step %Q/I perform the :kibana_index_pattern_exist web action with:/,table(%{
+        | index_pattern_name | #{pattern_name} |
+    })
+    unless @result[:success]
+        step %Q/I run the :go_to_kibana_management_page web action/
+        step %Q/I perform the :create_index_pattern_in_kibana web action with:/, table(%{
+            | index_pattern_name | "#{pattern_name}" |
+         })
+         raise "Failed to create index pattern #{pattern_name}" unless @result[:success]
+    end
+end
+
+Given /^cluster-admin create the #{QUOTED} pattern for the #{WORD} user$/ do | pattern_name, who |
+    user(word_to_num(who))
+    user_name=user.name
+    user_token=user.cached_tokens.first
```

Comment:
@anpingli Should these lines under `step %Q/I switch to cluster admin pseudo user/` ?